project_1/src/simulator.cpp

```diff
@@ -0,0 +1,64 @@
+#include "simulator.hpp"
+
+#include <chrono>
+
+NetworkSimulator::NetworkSimulator(const std::vector<std::uint32_t>& distances_to_receiver) : servers_number(distances_to_receiver.size()) {
+    for (std::uint32_t distance: distances_to_receiver) {
+        AddServer(distance);
+    }
+    std::uint32_t server_receiver_id = 0;
+    server_receiver = ServerBase(server_receiver_id);
+}
+
+void NetworkSimulator::AddServer(std::uint32_t distance_to_receiver) {
+    if (servers.size() < servers_number) {
```

Comment:
Do we really need `servers_number`? It looks like `servers.size()` is sufficient to get the number of servers?